lib/resty/core/socket_tcp.lua

```diff
@@ -172,7 +173,11 @@ do
     local old_socket_tcp = ngx.socket.tcp
 
     function ngx.socket.tcp()
-        local sock = old_socket_tcp()
+        local ok, sock = pcall(old_socket_tcp)
```

Comment:
The commit which adds the pcall around here should be merged with d75e6f6c935810e0669e26309183d172fc571bbc, unless the intent is that this whole PR is collapsed into a single commit when merging.

Reply:
Yes, we usually do the squashing at merge time, to keep a reasonable review history.